### PR TITLE
Prevent locking down after reading invalid state content

### PR DESCRIPTION
**Pull request by mumoshu**
Ref #16

### CODE REVIEW

File: pkg/tfhelmfile/resource_release.go [hunk 1]
```diff
@@ -129,6 +130,10 @@ func resourceShellHelmfileRelease() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			KeyDiffError: {
```
Comment (andrewnazarov) [inline pkg/tfhelmfile/resource_release.go:133]: Could you please explain why it's set in release, but not in releaseSet. Maybe I'm missing something.

Comment (mumoshu) [reply pkg/tfhelmfile/resource_release.go:133]: Ah good catch. I had updated wrong place... Fixed in the last commit.

**Comment by andrewnazarov** [comment]
What the general idea behind these changes? To prevent a state creation? What about Create callback then? The state can be broken there.

**Comment by mumoshu** [comment]
@andrewnazarov The idea is preventing to persist the "invalid" state, where "invalid" means any subsequent `helmfile-diff` run against the state fails.

As you pointed out, we'd also need to add the same logic to the "create" callback, too.

**Comment by mumoshu** [comment]
I ran a few manual tests and it turned out that it isn't an issue on create. Any helmfile failure on resource creation seems to not change the tfstate.

**Comment by mumoshu** [comment]
I confirmed that running `terraform apply` TWICE after fixing helmfile.yaml works with this patch. Not sure why just running it once after the fix isn't enough.

According to trace logs, it seems like `terraform apply` after fixing helmfile.yaml still fails due to that it results in a resource `Update` operation with the helmfile.yaml BEFORE the fix 🤔 The state is updated with the latest helmfile.yaml after the update failure though, which makes the second `terraform apply` run successful.

**Comment by mumoshu** [comment]
The need for running apply twice to fix the state was definitely due to a bug.
Now it's perfect!

**Pull request merged**